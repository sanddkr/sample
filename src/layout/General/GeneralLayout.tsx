import * as React from 'react';

import { GeneralLayoutProps } from './GeneralLayout.props';


export const GeneralLayout = (props:GeneralLayoutProps) => {
    return (
        <div className="row">
            <div className="col-sm-5 offset-sm-1">
                {props.children}
            </div>
            <div className="col-sm">
                <img src={props.image.src} alt={props.image.alt} />
            </div>
        </div>
    )
};