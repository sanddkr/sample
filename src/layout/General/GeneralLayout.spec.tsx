import * as React from "react";
import {mount, shallow} from "enzyme";
import * as renderer from 'react-test-renderer';
import GeneralLayout from './';
import { GeneralLayoutProps } from './GeneralLayout.props';

describe("Footer Component", () => {
    let wrapper, mountwrap;
    let layoutProps:GeneralLayoutProps = {
        children: <h1>Hi How are you</h1>,
        image: {
            src: '',
            alt: ''
        }
    }
    beforeEach(() => {
        wrapper = shallow(<GeneralLayout children={layoutProps.children} image={layoutProps.image} />);
        mountwrap = mount(<GeneralLayout children={layoutProps.children} image={layoutProps.image} />);
    });

    it('Footer Component renders correctly', () => {
        expect(wrapper.children().length).not.toBeLessThan(1);
    });

    it('Footer Component renders correctly using snapshots', () => {
        const rendered = renderer.create(
            <GeneralLayout children={layoutProps.children} image={layoutProps.image} />
        );
        expect(rendered.toJSON()).toMatchSnapshot();
    });
});