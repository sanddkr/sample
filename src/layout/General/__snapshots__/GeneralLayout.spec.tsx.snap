// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Footer Component renders correctly using snapshots 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-5 offset-sm-1"
  >
    <h1>
      Hi How are you
    </h1>
  </div>
  <div
    className="col-sm"
  >
    <img
      alt=""
      src=""
    />
  </div>
</div>
`;
